pkg/loadbalancer/target/metrics.go

```diff
@@ -253,14 +327,15 @@ func (hm *HitsMetrics) Collect() error {
 // the metrics/counter as value
 func (hm *HitsMetrics) getMetrics() (map[int]*expr.Counter, error) {
 	counters := map[int]*expr.Counter{}
+	conn := &nftables.Conn{}
 
-	rules, err := hm.getRules()
-	if err != nil {
-		return nil, err
-	}
+	for id, chain := range hm.fwmarkChains {
+		rules, err := conn.GetRules(hm.table, chain)
+		if err != nil || len(rules) < 1 || len(rules[0].Exprs) < 1 {
+			continue
+		}
 
-	for id, rule := range rules {
-		counterExpr := rule.Exprs[2].(*expr.Counter)
+		counterExpr := rules[0].Exprs[0].(*expr.Counter)
```

Comment:
Wouldn't it be safer to use type assertion testing instead? (i.e. checking for the second return value of type assertion)